### ISSUE TITLE
Myelin (Error 716 destroying event '0x3204a2e0'

### ISSUE THREAD

**Issue by xingjinglu** [NVIDIA/TensorRT#2391]
## Environment

**TensorRT Version**: 8.2.2.1
**NVIDIA GPU**: T4
**NVIDIA Driver Version**: 450.102.04
**CUDA Version**: 10.2
**CUDNN Version**:  8.2.4
**Operating System**:  centos7.2
**Python Version (if applicable)**: 3.8
**Tensorflow Version (if applicable)**: No
**PyTorch Version (if applicable)**: No
**Baremetal or Container (if so, version)**: 

### Case 1, when add --refit option, it will fail
trtexec --onnx=refit_500.onnx  --explicitBatch --workspace=10000000 --refit

### Case 2.  Success
trtexec --onnx=refit_500.onnx  --explicitBatch --workspace=10000000 

## Steps To Reproduce

10/14/2022-20:06:59] [I] Finish parsing network model
[10/14/2022-20:07:00] [I] [TRT] [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +4294222023, GPU +70, now: CPU 4294222436, GPU 1442 (MiB)
[10/14/2022-20:07:00] [I] [TRT] [MemUsageChange] Init cuDNN: CPU +137, GPU +88, now: CPU 4294222573, GPU 1530 (MiB)
[10/14/2022-20:07:00] [I] [TRT] Local timing cache in use. Profiling results in this builder pass will not be stored.
[10/14/2022-20:07:03] [E] Error[1]: [graphContext.h::~MyelinGraphContext::39] Error Code 1: Myelin (Error 716 destroying event '0x2df914a0'.)
[10/14/2022-20:07:03] [E] Error[1]: [resizingAllocator.cpp::deallocate::100] Error Code 1: Cuda Runtime (misaligned address)
[10/14/2022-20:07:03] [E] Error[1]: [resizingAllocator.cpp::deallocate::100] Error Code 1: Cuda Runtime (misaligned address)
[10/14/2022-20:07:03] [W] [TRT] Skipping tactic 0 due to Myelin error: 'final_sync': synchronize failed (716)
[10/14/2022-20:07:03] [E] Error[1]: [resizingAllocator.cpp::deallocate::100] Error Code 1: Cuda Runtime (misaligned address)
[10/14/2022-20:07:03] [E] Error[10]: [optimizer.cpp::computeCosts::2011] Error Code 10: Internal Error (Could not find any implementation for node {ForeignNode[Greater_17/y:0 + (Unnamed Layer* 3) [Shuffle]...y_piandan_pred]}.)
[10/14/2022-20:07:03] [E] Error[2]: [builder.cpp::buildSerializedNetwork::609] Error Code 2: Internal Error (Assertion enginePtr != nullptr failed. )
[10/14/2022-20:07:03] [E] Engine could not be created from network
[10/14/2022-20:07:03] [E] Building engine failed
[10/14/2022-20:07:03] [E] Failed to create engine from model.
[10/14/2022-20:07:03] [E] Engine set up failed

**Comment by xingjinglu** [comment]
Two problems:
1）when add --refit option, it failed, what is reason?
2) when add --refit, the engine size is smaller, why?
3) what does --refit really do, are there some docs about this?

**Comment by zerollzeng** [comment]
@BowenFu ^ ^ can you help here?

**Comment by Galaxygray** [comment]
I meet the same problem, when i add --refit, It's ok, but when I remove refit, It failed.

**Comment by BowenFu** [comment]
@Galaxygray Can you share your model or provide a minimal repro model?

**Comment by Galaxygray** [comment]
@BowenFu I‘m not very convenient to provide model, emmmmm
Could you give some advice to figure out where the wrong is

**Comment by BowenFu** [comment]
> @BowenFu I‘m not very convenient to provide model, emmmmm Could you give me some advice to figure out where the wrong is, thanks!

Hard to say before seeing the log. With `--refit` trtexec will build a refittable engine while without that option the engine would non-refittable. The graph optimization can be different for the two cases.

**Comment by Galaxygray** [comment]
@BowenFu Log is same as the issues，I'm interested in why refitter will affect graph optimization，What perfermance would be better with refitter or not

**Comment by BowenFu** [comment]
> @BowenFu Log is same as the issues，I'm interested in why refitter will affect graph optimization，What perfermance would be better with refitter or not

Perf would be better without --refit.

**Comment by Galaxygray** [comment]
@BowenFu  Thanks!, Another problem is when I remove refit, It's sucess, but when I loading engine in backup thread, It load meyline cost 300ms+ and locked the front doing inference thread.
<img width="1155" alt="image" src="https://user-images.githubusercontent.com/15278582/196423272-d811ea36-3560-47bf-a24e-368eae20b0c1.png">

**Comment by xingjinglu** [comment]
> > Three problems: 1）when add --refit option, it failed, what is reason? 2) when add --refit, the engine size is smaller, why? 3) what does --refit really do, are there some docs about this?
> 
> 1. The engine building failed with --refit option, this can be a TRT bug. Can you provide the onnx model?
> 2. Is the engine empty? I'm assuming that is case for failed engine building.
> 3. You can refer to this section. https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#refitting-engine-c
1. Now, I cann't provide the onnx model. But there are some details I can supply.
  a. There are two different docker images,  I.E. A and B,  there are same os, gcc, cuda libraries, but
  b. On docker A: it failed to build the engine with option --refit;
  c. on docker B: It is ok to build the engine with options --refit,  but I have to use docker A. 
3. yes， there are no engine, it is empty.
4.  thanks, I will read first.

**Comment by xingjinglu** [comment]
> @BowenFu It isn't empty, just decrease 2 times.

You means it will work with --refit sometimes, not always failed?
In my case, it always failed.

**Comment by BowenFu** [comment]
> @BowenFu Thanks!, Another problem is when I remove refit, It's sucess, but when I loading engine in backup thread, It load meyline cost 300ms+ and locked the front doing inference thread. <img alt="image" width="1155" src="https://user-images.githubusercontent.com/15278582/196423272-d811ea36-3560-47bf-a24e-368eae20b0c1.png">

Can you create a new issue for this performence problem?

**Comment by Galaxygray** [comment]
@BowenFu OK,Thanks!

**Comment by BowenFu** [comment]
> 1. Now, I cann't provide the onnx model. But there are some details I can share.
>    a. There are two different docker images,  I.E. A and B,  there are same version os, gcc, cuda,libc.so.6 libraries, but
>    b. On docker A: it failed to build the engine with option --refit;
>    c. on docker B: It is ok to build the engine with options --refit,  but I have to use docker A.
>    d. on the docker A,  it works for some batchsize(such as 100, 300, 200) and failed on other batch sizes( such as 500)
> 2. yes， there are no engine, it is empty. but I can get engine for some batchsize or on other dockers.
> 3. thanks, I will read first.

Are the two docker sharing the same release of TensorRT and are the two docker running on the same GPU? (And always try the latest release which may fix some issues.) It sounds strange to me if the two dockers behave differently while sharing the same software + hardware.

> BTW, the same bin codes(cuda libraries) run on different dockers, for some batchsize, it cause "Cuda Runtime (misaligned address)", it seems odd,
> what errors may lead to "Cuda Runtime (misaligned address)" ? Are there some tips for this?

If you building engines with dynamic-shape models, you need to set input shape ranges. Please refer to https://github.com/NVIDIA/TensorRT/tree/main/samples/trtexec#example-4-running-an-onnx-model-with-full-dimensions-and-dynamic-shapes.

**Comment by xingjinglu** [comment]
The problem is solved.
1）upgrade gcc to 7.3 on docker A, then trtexec can build the engine, but the python can't build the engine( conda, python-3.6.8)
 - I just try the python to build the engine, not try trtexec before.
2) Use the /usr/bin/python (version 3.6.8), the python can build the engine too. 

BTW, the reason why the conda-python failed is not clear.

**Comment by BowenFu** [comment]
@zerollzeng to close the issue since the problem is solved now.
